### PR TITLE
Add webp to screenshot options

### LINKED ISSUE #5348: Save Screenshots in WEBP Image Format
According to the 2.0 [documentation](https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#pagescreenshotoptions), a `page` instance has a `screenshot` method that accepts an object where you can set the `type` property (which is the image format) to either `png` or `jpeg`.

These days, instead of `png` or `jpeg` I try to use the `webp` image format as much as possible:
https://developers.google.com/speed/webp

Please add support for saving screenshots in the `webp` image format. This would also require making the `quality` option-property applicable to `webp` (in addition to its [currently exclusive] support for  `jpeg`), allowing you to set the quality of a webp screenshot.

### PR DESCRIPTION
**Pull request by jonchurch**
closes: #5348

Not sure since when, but webp is an option in [`Page.captureScreenshot`](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot) now.

All that is needed is to allow users to pass `type: "webp"` through. 

After making said change to allow passing webp through as an option, this code successfully creates webp screenshots using either `type: "webp"` or using the `.webp` extension from the filepath.
```js
const puppeteer = require('puppeteer')

async function run() {
const browser = await puppeteer.launch()
  const page = await browser.newPage()
  await page.setViewport({ width: 1280, height: 800 })
  await page.goto('https://www.nytimes.com/')
  await page.screenshot({ path: `${__dirname}/nytimes.webp`, fullPage: true, type: 'webp'})
  await page.screenshot({ path: `${__dirname}/nytimes_alt.webp`, fullPage: true,})
  await browser.close()
}

run().then(() => console.log('done')).catch(console.log)
```
Happy to take any help getting this mergeable, ie writing tests.

I'd upload a webp here but github doesn't support them!

### CODE REVIEW

**Comment by jonchurch** [comment]
@googlebot I signed it!

**Review by jschfflr — CHANGES_REQUESTED** [review]
Thanks for your contribution! The change looks good from a technical point of view but it would be great if you could also update the documentation (`docs/api.md`) and add a unit test (see `screenshot.spec.ts`). Thanks!

File: src/common/Page.ts [hunk 1]
```diff
@@ -157,7 +157,7 @@ export interface ScreenshotOptions {
   /**
    * @defaultValue 'png'
    */
-  type?: 'png' | 'jpeg';
+  type?: 'png' | 'jpeg' | 'webp';
```
Comment (jschfflr) [inline src/common/Page.ts:160]: Can you use `Protocol.Page.CaptureScreenshotRequestFormat` here?

File: src/common/Page.ts [hunk 2]
```diff
@@ -2634,7 +2637,7 @@ export class Page extends EventEmitter {
   }
 
   private async _screenshotTask(
```
Comment (jschfflr) [inline src/common/Page.ts:2646]: Not sure how `shouldSetDefaultBackground` should behave for `webp` - could you check that?

File: src/common/Page.ts [hunk 3]
```diff
@@ -2552,7 +2552,9 @@ export class Page extends EventEmitter {
     // (i.e. as a temp file).
     if (options.type) {
       assert(
-        options.type === 'png' || options.type === 'jpeg',
+        options.type === 'png' ||
```
Comment (jschfflr) [inline src/common/Page.ts]: Is it somehow possible to have TypeScript make sure that this check is extensive?

**Review by jschfflr — CHANGES_REQUESTED** [review]

**Comment by jschfflr** [comment]
@googlebot I consent.

**Comment by jschfflr** [comment]
@jonchurch I already went ahead and addressed the outstanding comments to get this into the next release :)

**Review by jschfflr — APPROVED** [review]

**Comment by jonchurch** [comment]
@jschfflr Yay! Thank you for getting this over the finish line 🥇

**Comment by jschfflr** [comment]
Sure, happy to help!

**Comment by cezary13k** [comment]
> zamyka : #5348
> 
> Nie jestem pewien od kiedy, ale webp jest obecnie dostępną opcją [`Page.captureScreenshot`](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-captureScreenshot).
> 
> Wystarczy pozwolić użytkownikom przejść `type: "webp"`.
> 
> Po wprowadzeniu wspomnianej zmiany umożliwiającej opcjonalne przesyłanie protokołu webp, ten kod pomyślnie tworzy zrzuty ekranu webp, używając `type: "webp"`albo `.webp`rozszerzenia ze ścieżki pliku.
> 
> ```js
> const puppeteer = require('puppeteer')
> 
> async function run() {
> const browser = await puppeteer.launch()
>   const page = await browser.newPage()
>   await page.setViewport({ width: 1280, height: 800 })
>   await page.goto('https://www.nytimes.com/')
>   await page.screenshot({ path: `${__dirname}/nytimes.webp`, fullPage: true, type: 'webp'})
>   await page.screenshot({ path: `${__dirname}/nytimes_alt.webp`, fullPage: true,})
>   await browser.close()
> }
> 
> run().then(() => console.log('done')).catch(console.log)
> ```
> 
> Chętnie przyjmę każdą pomoc w doprowadzeniu tej funkcjonalności do scalania, np. w pisaniu testów.
> 
> Chciałbym wgrać plik webp tutaj, ale github go nie obsługuje!

**Pull request merged**